lib/kubernetes-deploy/options_helper.rb

```diff
@@ -2,25 +2,58 @@
 
 module KubernetesDeploy
   module OptionsHelper
-    def self.default_and_check_template_dir(template_dir)
-      if !template_dir && ENV.key?("ENVIRONMENT")
-        template_dir = "config/deploy/#{ENV['ENVIRONMENT']}"
+    class OptionsError < StandardError; end
+
+    STDIN_TEMP_FILE = "from_stdin.yml.erb"
+    class << self
+      def with_consolidated_template_dir(template_dirs)
+        if template_dirs.length > 1 || template_dirs.include?('-')
+          Dir.mktmpdir("kubernetes-deploy") do |dir|
+            populate_temp_dir(temp_dir: dir, template_dirs: template_dirs)
+            yield dir
+          end
+        else
+          yield default_template_dir(template_dirs.first)
+        end
       end
 
-      if !template_dir || template_dir.empty?
-        puts "Template directory is unknown. " \
-          "Either specify --template-dir argument or set $ENVIRONMENT to use config/deploy/$ENVIRONMENT " \
-        + "as a default path."
-        exit(1)
+      def revision_from_environment
+        ENV.fetch('REVISION') do
+          puts "ENV['REVISION'] is missing. Please specify the commit SHA"
+          exit 1
+        end
       end
 
-      template_dir
-    end
+      private
+
+      def default_template_dir(template_dir)
+        if ENV.key?("ENVIRONMENT")
+          template_dir = "config/deploy/#{ENV['ENVIRONMENT']}"
+        end
+
+        if !template_dir || template_dir.empty?
+          raise OptionsError, "Template directory is unknown. " \
+            "Either specify --template-dir argument or set $ENVIRONMENT to use config/deploy/$ENVIRONMENT " \
+            "as a default path."
+        end
+
+        template_dir
+      end
 
-    def self.revision_from_environment
-      ENV.fetch('REVISION') do
-        puts "ENV['REVISION'] is missing. Please specify the commit SHA"
-        exit 1
+      def populate_temp_dir(temp_dir:, template_dirs:)
+        template_dirs.each do |template_dir|
+          if template_dir == '-'
+            File.open("#{temp_dir}/#{STDIN_TEMP_FILE}", 'w+') { |f| f.print($stdin.read) }
```

Comment:
Everywhere you're hardcoding a `/`  in this file we should change to `File.join(path, pieces)` for compatibility reasons

Reply:
Should we still check `$stdin.tty?` like you were before?

Reply:
We _could_. I removed it here since that's the convention I see adopted by most command line tools. On the other hand, it's not great UX if the task hangs waiting for input if the user makes an error in the pipeline. 

I'm leaning towards raising an exception if STDIN is specified but attached to a tty